### PR TITLE
VAULT: Added additional checks for setting fees

### PR DESCRIPTION
**Pull request by coachchucksol**
Audit had outlined the following functions did not have a `fee < MAX_FEE_BPS` check
set_reward_fee_bps() - fixed in this PR
set_program_fee_bps() - fixed in this PR
set_withdrawal_fee_bps() - fixed in previous PR

all set fee functions now have this check

### CODE REVIEW

File: vault_core/src/vault.rs [hunk 1]
```diff
@@ -333,8 +334,13 @@ impl Vault {
         u16::from(self.program_fee_bps)
     }
 
-    pub fn set_program_fee_bps(&mut self, program_fee_bps: u16) {
+    pub fn set_program_fee_bps(&mut self, program_fee_bps: u16) -> Result<(), ProgramError> {
+        if program_fee_bps > MAX_FEE_BPS {
```
Comment (buffalu) [inline vault_core/src/vault.rs:338]: there's MAX_BPS and MAX_FEE_BPS. should just use one?

Comment (coachchucksol) [reply vault_core/src/vault.rs:338]: We should

File: vault_core/src/vault.rs [hunk 2]
```diff
@@ -454,8 +459,8 @@ impl Vault {
             return Err(ProgramError::MissingRequiredSignature);
         }
 
-        if deposit_fee_bps > MAX_FEE_BPS {
-            msg!("Deposit fee exceeds maximum allowed of {}", MAX_FEE_BPS);
+        if deposit_fee_bps > MAX_BPS {
+            msg!("Deposit fee exceeds maximum allowed of {}", MAX_BPS);
             return Err(ProgramError::InvalidArgument);
```
Comment (buffalu) [inline vault_core/src/vault.rs:464]: nit: vault fee cap exceeded?

Comment (buffalu) [reply vault_core/src/vault.rs:464]: i think can do .into for custom error

Comment (coachchucksol) [reply vault_core/src/vault.rs:464]: Done

**Review by ebatsell — APPROVED** [review]

**Review by buffalu — APPROVED** [review]

**Pull request merged**